src/Chapter2.hs

```diff
@@ -611,8 +640,12 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
-
+takeEven :: [a] -> [a]
+takeEven l = go l []
+  where go [] accum = accum
+        go [x] accum = accum ++ [x]
+        go (x:_:xs) accum = go xs (accum ++ [x])
```

Comment:
The same suggestion for `duplicate` can be applied here is well. If you can think, how to avoid adding elements to the end of the list, and instead add to the beginning, but get the same result, that would be amazing!

In general, it's always a good idea to try to replace `l ++ [x]` with `x : l` whenever possible, though, in some cases it might not be that trivial to achieve the same result. Remember the analogy with trains :bullettrain_side: :train: :train: :train: